### PR TITLE
Module1 pe

### PR DESCRIPTION
**Pull request by rohanraj14**
Tried stretched Goals but not able to get result for Sort the item list by status. Check the last commit for file session_item.py at line 16-18. Not sure what I am doing wrong here.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -4,9 +4,26 @@
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
 
+
 @app.route('/')
 def index():
-    return 'Hello World!'
+    return render_template('index.html', items=session.get_items())
+
+@app.route('/addit', methods=['GET', 'POST'])
```
Comment (Jonesey13) [inline app.py]: I'd split this into 2 separate methods (for GET & POST)

File: app.py [hunk 2]
```diff
@@ -4,9 +4,26 @@
...
+def addit():
```
Comment (Jonesey13) [inline app.py]: I'd use the name `add_todo` for clarification

File: session_items.py [hunk 3]
```diff
@@ -13,7 +13,9 @@ def get_items():
     Returns:
         list: The list of saved items.
     """
-    return session.get('items', _DEFAULT_ITEMS)
+    # return session.get('items', sorted(_DEFAULT_ITEMS, key=lambda item: item['status'], reverse=True))
+    _SORTED_ITEMS = _DEFAULT_ITEMS.sort(key = lambda item : item['status'], reverse=True)
```
Comment (Jonesey13) [inline session_items.py]: This only sorts the default items (i.e. before anything is added to the flask session cookie)

File: templates/index.html [hunk 4]
```diff
@@ -8,10 +8,27 @@ <h1 class="display-4">To-Do App</h1>
   </div>
 
   <div class="row">
+    <form action="/addit" method="post">
+      <div class="col-md-12 form-group row">
+        <div class="input-group input-group-sm mb-3">
+          <div class="input-group-prepend">
+            <span class="input-group-text" id="inputGroup-sizing-sm">New Item</span>
+          </div>
+        <input class="form-control" type="text" name="new_item" id="item"/>
```
Comment (Jonesey13) [inline templates/index.html]: I'd add a `required` attribute here to prevent adding empty todos

File: templates/index.html [hunk 5]
```diff
@@ -8,10 +8,27 @@ <h1 class="display-4">To-Do App</h1>
...
+        <button class="btn btn-primary" type="submit" title="Add Item">Add It</button>
+      </div>
+      
+    </form>
     <div class="col-md-12">
       <h2>Items</h2>
       <ul class="list-group mb-4">
         <!-- List the todo items here -->
+        {% for item in items %} 
+          <li class="list-group-item {% if item['status'] == 'Complete' %}list-group-item-success{% endif %}">ID : {{item['id']}}, Title : {{item['title']}}, Status : {{item['status']}}
```
Comment (Jonesey13) [inline templates/index.html]: I'd split these elements over multiple lines to improve readability

**Review by Jonesey13 — APPROVED** [review]
Looks to all be working to me!

I've left some comments around clarity / refactoring but feel free to merge this PR in whenever you are ready. 👍

File: app.py (hunk 1 above)
Comment (rohanraj14) [reply app.py]: removed GET as not required

File: app.py (hunk 2 above)
Comment (rohanraj14) [reply app.py]: Changed name.

File: session_items.py (hunk 3 above)
Comment (rohanraj14) [reply session_items.py]: Moved sort function to app.py

File: templates/index.html (hunk 5 above)
Comment (rohanraj14) [reply templates/index.html]: I tried formatting

**Pull request merged**